waku/node/v2/wakunode2.nim

```diff
@@ -192,14 +192,8 @@ proc query*(w: WakuNode, query: HistoryQuery): HistoryResponse =
   ##
   ## Status: Not yet implemented.
   ## TODO Implement as wrapper around `waku_store` and send RPC.
-  result.messages = newSeq[Message]()
-
-  for msg in w.messages:
-    if msg[0] notin query.topics:
-      continue
-
-    # XXX Unclear how this should be hooked up, Message or WakuMessage?
-    # result.messages.insert(msg[1])
+  # DOES NOT SEEM TO WORK?
+  # w.wakuStore.query(query)
```

Comment:
The reason that this isn't working is because `init` is incorrectly overridden. The proper signature is `method init*(p: LPProtocol) {.base, gcsafe.} = discard`, so `WakuStore` and all other libp2p protocols need to be setup as:

```nim
method init*(ws: WakuStore) =
 
  proc handle(conn: Connection, proto: string) {.async, gcsafe, closure.} =
    var message = await conn.readLp(64*1024)
    var rpc = HistoryQuery.init(message)
    if rpc.isErr:
      return

    info "received query"

    let res = ws.query(rpc.value)

    await conn.writeLp(res.encode().buffer)

  ws.handler = handle
  ws.codec = WakuStoreCodec
  result = ws
```

Reply:
@dryajov tried what you mentioned. Does not seem to work.